### DISCUSSION TITLE
Why the `-file` build option?

### DISCUSSION THREAD

**Discussion by programandala-net** [odin-lang/Odin#2393] [Q&A] [answered]
Since I started learning Odin, I wonder about the `-file` build option. Why not just `odin build <file>` in order to treat a file as a self-contained package?

**Comment by Kelimion** [comment, marked as answer]
This is kind of answered when you use `odin help build`. Essentially, before `-file`, beginners would have `dir\a.odin` and `dir\b.odin`, build one of them and then complain that they didn't see declarations from the other file, despite telling Odin to build a single-file package.

It happened so many times we made it a dedicated option to make you think about the implication of building a single file.
```
>odin help build
odin is a tool for managing Odin source code
Usage:
        odin build [arguments]

        build   Compile directory of .odin files as an executable.
                One must contain the program's entry point, all must be in the same package.
                Use `-file` to build a single file instead.
                Examples:
                        odin build .                    # Build package in current directory
                        odin build <dir>                # Build package in <dir>
                        odin build filename.odin -file  # Build single-file package, must contain entry point.

        Flags

        -file
                Tells `odin build` to treat the given file as a self-contained package.
                This means that `<dir>/a.odin` won't have access to `<dir>/b.odin`'s contents.
```

**Reply by Kelimion** [reply]
That's to say it isn't a technical reason. Odin did originally build single-file packages without the flag just fine. The reason is social, because without the flag too many newcomers were confused about files in the same directory not being able to access each other's contents, having not yet really understood what it means to be a package.

**Reply by programandala-net** [reply]
Thank you. I saw no technical reason for `-file`, but maybe I was missing something.

I understand the "social" reason, but I think it's unfortunate to force everybody to use always an _unnecessary_ option just to prevent _initial_ errors by newcomers that didn't understood the documentation, which is clear.
